accepted/2020/workloads/workloads.md

```diff
@@ -0,0 +1,92 @@
+
+The set of TFM to workload mappings will be finite, and defined by the workloads. The SDK will carry a copy of this mapping, but will defer to information carried as part of installed workloads when they present newer information.
+
+### Acquisition
+
+Workloads will be acquired in two primary ways: via a supported installation tool or the .NET CLI.
+
+Note: The term "supported installation tool" is intended to mean an installation system that exposes components and manages their dependencies. For example, Visual Studio Installer, Docker, and APT do that, while MSI and PKG don't.
+
+Install tools must follow these rules for workloads it installs:
+
+* The workload manifest is included as a required dependency for all variants of the workload (some packs might be optional).
+* All referenced packs are included, matching the version specified in the workload manifest.
+
+Users should be able to install workloads via an install tool without any sense that there is a compositional model at play.
+
+The .NET CLI will expose a more extensive set of commands for interacting with workloads. The CLI may allow installing more than just workloads, such that the noun it exposes is not `workload` but something more general (TBD; `bundle` is used as the stand-in for a better term in the following examples). In this parlance, a workload is one kind of bundle.
+
+* `dotnet bundle install [bundle]` -- Installs a bundle by name.
+* `dotnet bundle restore [project]` -- Installs one or more workloads in terms of the dependent workloads found in one or more referenced projects. One can think of restore installing the default bundle for a workload. The workload may expose other opt-in bundles.
+* `dotnet bundle update [bundle | project]` -- Updates a bundle, in terms of a project, by name, or as part of updating all known bundles.
+
+We don't intend for the .NET CLI-based installation to be mix-and-matched with a package tool (like APT). Instead, we are thinking that an installation tool would provide a mapping file that mapped its packages to workloads, so that `dotnet bunble restore` could provide meaningful error messages that direct users to install workloads via the installation tools. We still need to spend more time defining interactions between the .NET CLI and an installation tools.
+
+In addition to being optional, workloads will be updatable. The SDK will include a mechanism to discover if a workload manifest pack has been updated. An updated manifest could represent bug fixes or new features (like updated bindings for an operating system update). If a manifest has been updated, the user will be notified, and have the opportunity to acquire the new version. Updates will not be automatic. This is roughly similar to the combination of `apt-get update` and `apt-get upgrade` with the [Debian package manager](https://en.wikipedia.org/wiki/APT_(software)).
```

Comment:
What does "In addition to being optional, workloads..." mean. If a TFM demands a workload, it isn't optional (rest of sentence - they are updatable - is good).

Reply:
I'm not sure it's clear enough at this point to the reader that the manifest installed on disk reflects the workload on disk. "Updating the manifest" is a rather trivial sounding thing, which I think will make this paragraph confusing. I think some tweaks here will have good results.

Reply:
Well, this whole feature is about optional workloads, relative to the SDK. I think of workloads as having two main characteristics: optional and updatable. In some cases (like for ASP.NET Core templates), we might only use the updatable part. Fair?

Reply:
Simplified the wording.